src/ropes/macros.rs

```diff
@@ -95,3 +95,88 @@ macro_rules! impl_rope {
         }
     }
 }
+
+
+/// Generate struct definition and implementation of RopeSlice
+///
+/// This is done through a macro because it relies on the Lnode type which is different
+/// depending on which type of Rope you are working with. This macro is used in both the
+/// rope & src_rope modules of this crate. In both places, the Lnode type from that
+/// module is passed in so we can generate the correct struct definition.
+macro_rules! generate_ropeslice_struct {
+    ($lnode: ty) => {
+        // A view over a portion of a Rope. Analagous to string slices (`str`);
+        pub struct RopeSlice<'rope> {
+            // All nodes which make up the slice, in order.
+            nodes: Vec<&'rope $lnode>,
+            // The offset of the start point in the first node.
+            start: usize,
+            // The length of text in the last node.
+            len: usize,
+            // The index of the current byte - only used for iterating
```

Comment:
Oh, I see you implement Iterator for RopeSlice. It is probably better to have a separate RopeSliceIterator